### DISCUSSION TITLE
Trigger fetch all on open Git Extension and change of repository

### DISCUSSION THREAD

**Discussion by 1Jesper1** [gitextensions/gitextensions#11148] [Q&A] [answered]
### Feature description

Trigger fetch all on open Git Extension and change of repository

Sometimes a developer forgets to fetch and pull the branches. Would be useful to have to option to do this automatically.

### Environment

- Git Extensions 4.1.0.16698
- Build 1fe52a137acef1fbd2b653dea263b44ad5844177
- Git 2.40.1.windows.1
- Microsoft Windows NT 10.0.22621.0
- .NET 6.0.21

**Comment by vbjay** [comment]
Could do it in a script if we have that event. A before repo switch event. But the other thing is you can setup background fetch.

**Comment by vbjay** [comment, marked as answer]
> Could do it in a script if we have that event. A before repo switch event. But the other thing is you can setup background fetch.

![image](https://github.com/gitextensions/gitextensions/assets/3629489/5fe1625e-8c6b-49e3-af4b-0af2580a74aa)

**Reply by RussKie** [reply]
I can think of many scenarios where this blanket setting would be quite undesirable, e.g., working offline or switching between large and busy repositories (e.g., fetching the Visual Studio repo takes minutes). 
As @vbjay suggested above you can enable per-repo auto-fetch with the plugin.
